src/rgw/rgw_basic_types.h

```diff
@@ -255,6 +256,7 @@ struct RGWUploadPartInfo {
   uint64_t size;
   uint64_t accounted_size{0};
   std::string etag;
+  rgw::cksum::Cksum cksum;
```

Comment:
looks like I did update RGWUploadPartInfo, following imtz's lead.  I kind of favor this, even if we don't immediately implement a consistency check?

Reply:
i think we need this to calculate the checksums for multipart uploads in `RadosMultipartUpload::complete()`. that has access to the list of `RGWUploadPartInfo`s, but doesn't read each part object for xattrs like RGW_ATTR_AMZ_CKSUM